### ISSUE TITLE
Bounds not adjusted when rotating device

### ISSUE THREAD

**Issue by Farasy** [alexvasilkov/GestureViews#112]
Dear Alex, rotating the device will not re-adjust a photo to fit the screen.
Instead you have to zoom out to make it fit.

`
image.getController().getSettings().setFillViewport(true);
image.getController().getSettings().setOverzoomFactor(10f);
 image.getController().getSettings().setDoubleTapZoom(2f);
 image.getController().getSettings().setMaxZoom(20f);
 image.getController().getSettings().setFitMethod(Fit.INSIDE);
`

Manifest
`
android:configChanges="orientation|keyboardHidden|screenSize|locale"
`

**Comment by alexvasilkov** [comment]
You can reset image state on orientation change manually with:
```java
image.getController().resetState();
```
Is it what you are looking for?

**Comment by Farasy** [comment]
Thanks, Alex.

**Comment by Farasy** [comment]
`resetState()` nor `updateState()` forces the view to rescale to fit the screen without crop.

`@Override public void onConfigurationChanged(Configuration newConfig){
        super.onConfigurationChanged(newConfig);
com.alexvasilkov.gestures.commons.RecyclePagerAdapter.ViewHolder holder = mPagerAdapter.getViewHolder(mPager.getCurrentItem());
com.alexvasilkov.gestures.views.GestureImageView iv = PhotoPagerAdapter.getImage(holder);
iv.getController().resetState();
}`

A simple redraw should do it.

**Comment by Farasy** [comment]
Got it.

Setting `setMaxZoom()` to `1f` before or after `resetState()` / `updateState()` and back to required max afterwards.

**Comment by XJIOP** [comment]
my solution
```
    private boolean resetZoom = false;

    @Override
    public void onConfigurationChanged(Configuration newConfig){
        super.onConfigurationChanged(newConfig);

        resetZoom = true;
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, final ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_image, container, false);

        final GestureImageView gestureImageView = view.findViewById(R.id.view_image);
        final GestureControllerForPager controller = gestureImageView.getController();

        controller.addOnStateChangeListener(new GestureController.OnStateChangeListener() {
            @Override
            public void onStateChanged(State state) {
                if(resetZoom) {
                    resetZoom = false;
                    controller.resetState();
                }
            }
            @Override
            public void onStateReset(State oldState, State newState) {}
        });
    } 
```

**Comment by alexvasilkov** [comment]
@Farasy Is this issue solved for you?

**Comment by Farasy** [comment]
Looks like my workaround fixes the issue on most devices but some users report that a photo disappears on rotate.

**Comment by Farasy** [comment]
Did you change the behaviour on 2.5.1?

**Comment by alexvasilkov** [comment]
Not sure what behaviour changes you are referring to, there were no changes related to state reset.

I reproduced your issue myself, the easiest fix for me was:

```java
image.getViewTreeObserver().addOnGlobalLayoutListener(() -> image.getController().resetState());
```

I will think about fixing it within the library in next versions, will need to consider implications first.

**Comment by alexvasilkov** [comment]
Fixed in v2.5.2
